Added option to not show back face of text

Danielbook opened the pull request:
Added an optional option to textcomponents, option.backface {bool}. 
The extrusion of the text is set to 0 if user don't want back facing text

Danielbook committed:
Added option to not show back face of text

Added an optional option to textcomponents, option.backface {bool}. 
The extrusion of the text is set to 0 if user don't want back facing text

spotscale commented on src/goo/geometrypack/text/TextMeshGenerator.js:
```diff
@@ -400,24 +401,28 @@ function meshesForText(text, font, options) {
 			meshBuilder.addMeshData(meshData, transform);
 		}
+	
+		if (options.backface)
```
Please add { }

Danielbook committed:
Added {} around if-else statements

rherlitz commented on src/goo/geometrypack/text/TextMeshGenerator.js:
```diff
@@ -402,11 +402,11 @@ function meshesForText(text, font, options) {
-		  frontFace();
-      		else // If the back face shouldn't be visible, set extrusion to 0s
+		if (options.backface) {
+			frontFace();
```
Shouldnt it be the backface()  that is drawn if options.backface is true?

Also, could you make the default behavior be to draw backface if nothing is specified (to behave like before), by doing options.backface = options.backface !== undefined ? options.backface || true; at the top?

PabloK replied:
I agree with @rherlitz. Thank you for the review!

Danielbook replied:
Thanks for the review, added the options.backface at the top, but when i change that the backface() is to be drawn when option.backface is true, I get the text backwards. The text isn't visible on the other side of the material due to the other face not being drawn.

With frontface drawn, no backface:
![skarmavbild 2016-07-13 kl 10 26 32](https://cloud.githubusercontent.com/assets/6060731/16796708/58649322-48e4-11e6-8a21-321e56fbc5df.png)

With backface drawn, no frontface:
![skarmavbild 2016-07-13 kl 10 25 50](https://cloud.githubusercontent.com/assets/6060731/16796776/9f20a148-48e4-11e6-9763-c0a45e70eb55.png)

Danielbook committed:
Update TextMeshGenerator.js

Danielbook committed:
Updated the functions frontface() and backface()

Updated the functions to draw correct face of glyph mesh

Danielbook commented:
Think the geometry pack now is correct, and the option to show backface is true as default.

Danielbook committed:
Fixed bug

For some reason, the backface has to be drawn before front face, or else the text gets all weird

Danielbook commented:
The bug was because of the extrusion, fixed now!

Should be good to go 👍
